app/2.0/docs/devguide/dom-template.md

```diff
@@ -94,20 +94,30 @@ important rules:
 
 -   Don't modify the superclass template in place; make a copy before modifying.
 -   If you're doing anything expensive, like copying or modifying an existing template,
-    you should memoize the modified template so you don't have to regenerate it when the
+    you should memoize the modified template so you don't have to regenerate it each time the
     getter is called.
+-   Consider that the superclass template getter could return either a string _or a `<template>` element.
 
 The following example shows a simple modification based on a parent template:
 
 ```js
-(function() {
+(function() { 
   let memoizedTemplate;
 
-  class MyExtension extends MySuperClass {
+  class SimpleExtension extends MySuperClass {
     static get template() {
       if (!memoizedTemplate) {
-        // create a clone of superclass template (`true` = "deep" clone)
-        memoizedTemplate = MySuperClass.template.cloneNode(true);
+        // get the superclass template, which could be 
+        // either a string or a template element
+        const superTemplate = MySuperClass.template;
+        if (typeof superTemplate == 'string') {
```

Comment:
@sorvell @justinfagnani Have you discussed and/or hit this issue yet in 2.0 → 3.0 migration?  Post-`modularizer`, any subclassers that extended the super template under the assumption that it was an `HTMLTemplateElement` would be broken unless they used this type of super-defensive code (which I'm sure most would not have).  Most of our elements wouldn't have hit this yet since they're all still hybrid, so that won't be a good litmus test.  Seems like we kinda screwed up allowing a subclassable field to have an ambiguous type. 🤦‍♂️

Maybe you've already discussed this and have a plan? They'll be 3.0 elements so maybe breaking subclassers is just ok?

Is it worth considering having `modularizer` do a safer thing by generating code that creates a template element rather than just a string:

```js
static get template() {
  return makeTemplate(`<-- (template html) -->`);
}
```

Or we could go the other way and make a breaking change to 3.0 by saying template strings have to be defined via e.g. `static get templateString()`, and `static get template()` is always a `HTMLTemplateElement` (and `modularizer` would generate `templateString`).

Reply:
@kevinpschaaf @sorvell can we chat about this in person?